<launch>
  <!-- nodelet name is /camera_nodelet_manager -->
  <arg name="manager" default="/camera/camera_nodelet_manager" />
  <param name="depth_registered" value="true"/>
  <include file="$(find openni_launch)/launch/openni.launch">
    <arg name="depth_registration" value="true"/>
  </include>
  <arg name="input_point" default="/camera/depth_registered/points" />

  <group ns="pcl_nodelet">
     <node pkg="nodelet" type="nodelet"
          name="voxelgrid"
          args="load pcl/VoxelGrid $(arg manager)"
          output="screen" clear_params="true">
      <remap from="~input" to="$(arg input_point)" />
      <rosparam>
        filter_field_name: z
        filter_limit_min: 0.0
        filter_limit_max: 5.0
        leaf_size: 0.01
      </rosparam>
    </node>
    <node pkg="nodelet" type="nodelet"
          name="particle_filter_tracker"
          args="load jsk_pcl/ParticleFilterTracking $(arg manager)"
          output="screen" clear_params="true">
      <remap from="~input" to="/pcl_nodelet/voxelgrid/output" />
    </node>
  </group>

  <node pkg="jsk_pcl_ros" name="renew_trakcing" type="renew_trakcing.py" />

  <node pkg="rviz" type="rviz" name="rviz" args=" -d $(find jsk_pcl_ros)/sample/rviz/tracking.rviz"/>
</launch>
